scene/resources/gradient.cpp

```diff
@@ -225,3 +230,22 @@ Color Gradient::get_color(int pos) {
 int Gradient::get_point_count() const {
 	return points.size();
 }
+
+void Gradient::initialize() {
+	_interpolation_gradient = Ref<Gradient>();
+	_interpolation_gradient.instantiate();
+	_interpolation_gradient->set_offset(0, 0.0f);
+	_interpolation_gradient->set_offset(1, 1.0f);
+}
+
+void Gradient::deinitialize() {
+	_interpolation_gradient = Ref<Gradient>(nullptr);
+}
+
+Color Gradient::interpolate(const Color &p_from, const Color &p_to, float p_weight, ColorSpace p_interpolation_color_space, InterpolationMode p_interpolation_mode) {
```

Comment:
Static method yet no thread safety in mind?

Reply:
I reverted the initialization code. Now, `_interpolation_gradient` is `thread_local`, so it should do the trick (i think).